azurerm/resource_arm_private_link_service.go

```diff
@@ -0,0 +1,415 @@
+	t := d.Get("tags").(map[string]interface{})
+
+	parameters := network.PrivateLinkService{
+		Location: utils.String(location),
+		PrivateLinkServiceProperties: &network.PrivateLinkServiceProperties{
+			AutoApproval: &network.PrivateLinkServicePropertiesAutoApproval{
+				Subscriptions: utils.ExpandStringSlice(autoApproval.List()),
+			},
+			Visibility: &network.PrivateLinkServicePropertiesVisibility{
+				Subscriptions: utils.ExpandStringSlice(visibility.List()),
+			},
+			IPConfigurations:                     expandArmPrivateLinkServiceIPConfiguration(primaryIpConfiguration),
+			LoadBalancerFrontendIPConfigurations: expandArmPrivateLinkServiceFrontendIPConfiguration(loadBalancerFrontendIpConfigurations),
+			//Fqdns:                                utils.ExpandStringSlice(fqdns),
+		},
+		Tags: tags.Expand(t),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating Private Link Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of Private Link Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving Private Link Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("API returns a nil/empty id on Private Link Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmPrivateLinkServiceRead(d, meta)
+}
+
+func resourceArmPrivateLinkServiceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Network.PrivateLinkServiceClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["privateLinkServices"]
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Private Link Service %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Private Link Service %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("location", azure.NormalizeLocation(*resp.Location))
+
+	if props := resp.PrivateLinkServiceProperties; props != nil {
+		d.Set("alias", props.Alias)
+		if props.AutoApproval != nil {
+			if err := d.Set("auto_approval_subscription_ids", utils.FlattenStringSlice(props.AutoApproval.Subscriptions)); err != nil {
+				return fmt.Errorf("Error setting `auto_approval_subscription_ids`: %+v", err)
+			}
+		}
+		if props.Visibility != nil {
```

Comment:
I think the glatten function will deal with nil

Reply:
Fixed.